malaria-ios/ViewControllers/UserProfileViewController.swift

```diff
@@ -0,0 +1,199 @@
+  func saveRemainingPills(textField: UITextField)
+}
+
+// MARK: - Table View controller
+
+class UserProfileViewController: UIViewController {
+  
+  @IBOutlet weak var remindMeWeeksButton: UIButton!
+  @IBOutlet weak var tableView: UITableView!
+  @IBOutlet weak var tableViewHeightConstraint: NSLayoutConstraint!
+  @IBOutlet weak var remainingLabel: UILabel!
+  
+  private var medicine: [Medicine] = []
+  private var reminderValue: PillStatusNotificationsManager.ReminderInterval = .OneWeek {
+    didSet {
+      remindMeWeeksButton.setTitle(reminderValue.toString(), forState: .Normal)
+    }
+  }
+  
+  // Core Data
+  private var context: NSManagedObjectContext?
+  private var medicineManager: MedicineManager?
+  private var psnm: PillStatusNotificationsManager?
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    // Set background image
+    view.backgroundColor = UIColor(patternImage: UIImage(named: "background")!)
+    
+    // Get the last reminder value from the User Defaults
+    let value = UserSettingsManager.UserSetting.PillReminderValue.getString()
+    
+    remindMeWeeksButton.setTitle(value, forState: .Normal)
+
+    self.reminderValue = value.toReminderValue()
+  }
+  
+  override func viewWillAppear(animated: Bool) {
+    super.viewWillAppear(animated)
+    
+    context = CoreDataHelper.sharedInstance.createBackgroundContext()!
+    medicineManager = MedicineManager(context: context!)
+    psnm = PillStatusNotificationsManager(context: context!)
+    
+    getRegisteredMedicine()
+    calculateMedicineLeft()
+    recalculateTableHeight()
+  }
+  
+  func dismissKeyboard() {
+    view.endEditing(true)
+  }
+  
+  func getRegisteredMedicine() {
+    medicine = []
+    medicine.append(medicineManager!.getCurrentMedicine()!)
+    
+    // To get all medicine use only:
+    // medicine = medicineManager!.getRegisteredMedicines()
+  }
+  
+  func calculateMedicineLeft() {
+    guard let currentMedicine = medicineManager!.getCurrentMedicine() else {
+      remainingLabel.text = ""
+      return
+    }
+    
+    tableView.reloadData()
+
+    // remaining value is floored (opposed to ceiled)
+    let remaining = Int(currentMedicine.remainingMedicine) / currentMedicine.interval
+    
+    let timeMeasuringUnit = (currentMedicine.interval == 7) ? "weeks" : "days"
+    
+    remainingLabel.text = "You have \(remaining) \(timeMeasuringUnit) of \(currentMedicine.name) left."
+    
+    // Save pill status to show it in the widget
+    WidgetSettingsManager.WidgetSetting.RemainingPills.setObject(remaining)
+    WidgetSettingsManager.WidgetSetting.RemainingPillsMeasuringUnit.setObject(timeMeasuringUnit)
+    
+    let shouldPresentNotification: Bool = psnm!.shouldPresentNotification(remaining, reminderValue: reminderValue.rawValue)
+    
+    shouldPresentNotification ? psnm!.scheduleNotification(NSDate()) : psnm!.unsheduleNotification()
+    
+    let defaultBrownTint = self.remindMeWeeksButton.titleLabel?.textColor
```

Comment:
Imo, this method does too many things. I've divided it into multiple methods in this commit that will come.